serde/src/private/de.rs

```diff
@@ -797,19 +797,13 @@ mod content {
     /// Used by generated code to deserialize an internally tagged enum.
     ///
     /// Not public API.
-    pub struct TaggedContent<'de, T> {
-        pub tag: T,
-        pub content: Content<'de>,
-    }
-
-    /// Not public API.
-    pub struct TaggedContentVisitor<'de, T> {
+    pub struct TaggedContentVisitor<T> {
         tag_name: &'static str,
         expecting: &'static str,
-        value: PhantomData<TaggedContent<'de, T>>,
+        value: PhantomData<T>,
     }
```

Comment:
I'm not really convinced that this is better than the status quo, so I'd prefer to just leave it as-is.

Reply:
The removed type is used only in one place for returning pair of values from one function, for that task a dedicated language construct is exist. Occam's razor says us that we should remove superfluous entity.

Reply:
I think either way is all right and justifiable for different reasons, but I am on board with this change.